spanner/value.go

```diff
@@ -1899,6 +2018,33 @@ func decodeValue(v *proto3.Value, t *sppb.Type, ptr interface{}, opts ...decodeO
 			return errBadEncoding(v, err)
 		}
 		proto.Unmarshal(y, p)
+	case *NullProto:
+		if p == nil {
+			return errNilDst(p)
+		}
+		if p.ProtoVal == nil {
+			return errNilDst(p.ProtoVal)
```

Comment:
Does the users sees an error msg like this:
cannot decode into nil type proto.Message.

The original context of "NullProto.ProtoVal is nil" is lost. Is it possible to improve this?

Reply:
It this also true for other error msgs we give?

Reply:
Updated the error message to below for better readability
`field ProtoMessageVal in spanner.NullProtoMessage cannot be nil`